text/0031-integ-testing.md

```diff
@@ -0,0 +1,542 @@
+snapshot, apptest will skip deployment and will not run any assertions. The test will be considered
+to have passed. This behaviour can be overridden either by setting the `ignoreSnapshot` option for
+individual tests or by setting the `CDK_APPTEST_IGNORE_SNAPSHOT` environment variable and will apply
+to all tests executed in that environment.
+
+However, when the synthesized cloud assembly does not match the snapshot, the test will be executed
+in full and the snapshot will be updated. As before, the changes to the snapshots must be checked into
+the source tree.
+
+[cloud assembly]: https://docs.aws.amazon.com/cdk/api/latest/docs/cloud-assembly-schema-readme.html#cloud-assembly
+
+### Test Execution & Report
+
+The `apptest` suite can be executed via any standard test runner available in the programming
+language of your choice - jest for javascript or typescript, junit for Java, pytest for Python, etc.
+
+As shown in the above section, it is recommended to use the `apptest` suffix to your test files
+to differentiate them from unit and other kinds of tests - `xxx.apptest.ts` in a typescript
+project, `xxx.apptest.java` for Java project, etc.
+
+This setup will enable defining a separate script that runs the apptest suite - npm script,
+Ant target, Maven phase, etc. - by using `*.apptest.*` as the file filter in your test runner.
+
+The test will be executed when the `run()` method is called on the `apptest.Test` class.
+This method will return a list of assertion failures (if any), that can then be used to notify
+the test runner of a test case failure.
+
+Test reports can be easily generated from the reporting mechanisms already available via these
+test runners.
+
+> RFC Note: See [Appendix B](#appendix-b---test-execution) for details on this design.
+
+### Testing AWS CDK Apps and construct libraries
+
+The industry standard practice is that app developers typically run pre-production instances
+of their applications or services as part of their deployment pipeline where integration tests
+are executed. This is no different for AWS CDK applications.
+
+In such cases, the apptest suite should define *ONLY* the tests and point them at the existing
+pre-production app instance, such as via a well-known endpoint or CloudFormation stack outputs.
+The apptest suite, in this case, will deploy (and towards the end destroy) a "testing" stack
+where the tests would be executed.
+
+The usage of `apptest` when testing construct libraries is different from that when it is used
+for apps.
+
+The `apptest` suite for construct libraries can be used to verify the functionality of the construct
+library when it is used with various valid configurations.
+Each `apptest` case would define a new AWS CDK `Stack` or `App` that includes the construct
+library, followed by assertions. Upon execution, both the app and the assertions will be deployed
+(and finally destroyed) as part of the test execution.
+
```

Comment:
Can you provide some examples here? I am not sure I understand what this section is trying to say